core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -258,12 +260,19 @@ public String analyzer() {
         return this.analyzer;
     }
 
+    @Deprecated
     public Boolean useAllFields() {
-        return useAllFields;
+        return fieldsAndWeights.size() == 1 && fieldsAndWeights.keySet().stream().anyMatch(Regex::isMatchAllPattern);
     }
 
+    /**
+     * This setting is deprecated, set {@link #field(String)} to "*" instead.
+     */
+    @Deprecated
     public SimpleQueryStringBuilder useAllFields(Boolean useAllFields) {
-        this.useAllFields = useAllFields;
+        if (useAllFields) {
+            this.fieldsAndWeights = Collections.singletonMap("*", 1.0f);
```

Comment:
It also breaks what I would expect to happen with:

```java
builder.useAllFields(true);
builder.field("foo", 1.0);
```

Where I would expect it to use `all_fields` mode, but instead it does not work since the order matters.